app/gather/api/views.py

```diff
@@ -142,21 +152,24 @@ def valid_header(name):
         # not hosted on that server. The webserver
         # should add the correct Host based on the request.
         # this problem might not be exposed running on localhost
+        headers.pop('Host', None)
 
-        if 'Host' in headers:  # pragma: no cover
-            del headers['Host']
-
-        # builds url with the query string
-        param_str = request.GET.urlencode()
-        url = request.path + ('?{}'.format(param_str) if param_str else '')
-
+        log(f'{method}  {url}')
         response = requests.request(method=method,
                                     url=url,
                                     data=request.body if request.body else None,
                                     headers=headers,
                                     *args,
                                     **kwargs)
-        return HttpResponse(response, status=response.status_code)
+        http_response = HttpResponse(response, status=response.status_code)
+        # copy from the original response headers the exposed ones
+        # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers
+        # https://fetch.spec.whatwg.org/#http-access-control-expose-headers
+        expose_headers = response.headers.get('Access-Control-Expose-Headers', '').split(', ')
```

Comment:
This looks like a good compromise - which header(s) is it that you need again?

Reply:
`Content-Disposition`

https://github.com/eHealthAfrica/aether/pull/336/files#diff-d2d8c6bbb578ea8da3c46269684afe0eR122